### PR TITLE
[fix](cluster key) fix some cluster key problems

### PR DESCRIPTION
**Pull request by mymeiyi**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

1. support randomly select mow cluster keys when `random_add_cluster_keys_for_mow=true` in `fe.conf`, used for test
2. fix bug of schema change: memtable use wrong column id
3. fix bug of compaction and schema change: if one segment add_columns multi times, the primary key index is only  sorted partially

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by mymeiyi** [comment]
run buildall

File: be/src/olap/rowset/vertical_beta_rowset_writer.cpp [hunk 1]
```diff
@@ -50,7 +50,8 @@ template <class T>
     requires std::is_base_of_v<BaseBetaRowsetWriter, T>
 Status VerticalBetaRowsetWriter<T>::add_columns(const vectorized::Block* block,
                                                 const std::vector<uint32_t>& col_ids, bool is_key,
-                                                uint32_t max_rows_per_segment) {
+                                                uint32_t max_rows_per_segment,
+                                                bool has_cluster_key) {
```
[the remark this answers is not in the document]
Comment (dataroaring) [reply be/src/olap/rowset/vertical_beta_rowset_writer.cpp:54]: And it would generates too many segments.

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by zhannngchen — APPROVED** [review]

**Pull request merged**